packages/playground/ssr-framework/vue-framework/createPageRender.js

```diff
@@ -0,0 +1,75 @@
+const { renderToString } = require('@vue/server-renderer')
+const { relative } = require('path')
+const { createSSRApp } = require('vue')
+const hydrateFile = require.resolve('./hydrate')
+const getPagesFile = require.resolve('./getPages')
+
+module.exports.createPageRender = createPageRender
+
+const env = {}
+
+function createPageRender(viteServer, root, isProduction) {
+  Object.assign(env, { viteServer, root, isProduction })
+  return render
```

Comment:
and here
... and theoretically more below :thinking:  
it's easier to read a file from top to bottom then from bottom to top :thinking:

Reply:
That's on purpose: the first two functions are `createPageRender()` and `render()` so that the reader knows "Ah okay, so this file exports a factory that creates a function that renders HTML". The reader gets to know the high level purpose of the file just by skimming over the first lines. If I were to put all low-level functions first, then I'm forcing him to go through all that low-level stuff before he achieves the high-level purpose of the file. As a reader I find it considerably more comfy to get a high-level impression first.

A reviewer wants to read the *entire* file, but a real reader usually only wants to get a high-level idea of what the file is about.

From that perspective, it also makes sense to put all exports on top of the file: it tells the reader what the file is about.